File: historian-core/src/main/java/net/yslibrary/historian/internal/LogEntity.java
package net.yslibrary.historian.internal;

/**
 * Entity class representing log
 */

public class LogEntity {
  public final String priority;
  public final String message;
  public final long timestamp;
  public final String tag;

  private LogEntity(String priority, String tag, String message, long timestamp) {
    this.priority = priority;
    this.tag = tag;
    this.message = message;
    this.timestamp = timestamp;
  }

  @SuppressWarnings("WeakerAccess")
  public static LogEntity create(int priority, String tag, String message, long timestamp) {
    return new LogEntity(Util.priorityString(priority), tag, message, timestamp);
  }
}

File: historian-core/src/main/java/net/yslibrary/historian/internal/LogEntity.kt
package net.yslibrary.historian.internal

/**
 * Entity class representing log
 */
data class LogEntity(
    @JvmField val priority: String,
    @JvmField val tag: String,
    @JvmField val message: String,
    @JvmField val timestamp: Long
) {
    companion object {
        @JvmStatic
        fun create(priority: Int, tag: String?, message: String, timestamp: Long): LogEntity =
            LogEntity(
                priority = priority.toPriorityString(),
                tag = tag.orEmpty(),
                message = message,
                timestamp = timestamp
            )
    }
}
